trees/trees.go

```diff
@@ -47,13 +47,36 @@ func FromContext(ctx context.Context) (*trillian.Tree, bool) {
 
 // GetOpts contains validation options for GetTree.
 type GetOpts struct {
-	// TreeType is the expected type of the tree. Use trillian.TreeType_UNKNOWN_TREE_TYPE to
-	// allow any type.
-	TreeType trillian.TreeType
+	// TreeTypes lists allowed types of the tree. If empty, any type is allowed.
+	TreeTypes []trillian.TreeType
```

Comment:
A map might be better again then you can just look up the type.

Reply:
Done. Not sure though about the performance. I'd bet a 2-element slice lookup would be faster.
Anyway, it's probably okay because this thing is never used heavily in loops.